src/components/settings/apiAccessTokens/CreateUpdateApiAccessToken.tsx

```diff
@@ -0,0 +1,343 @@
+import Tooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+import FileCopyIcon from '@material-ui/icons/FileCopy';
+import {
+  ErrorMessage,
+  Field,
+  FieldArray,
+  FieldArrayRenderProps,
+  Form,
+  Formik,
+} from 'formik';
+import { TextField } from 'formik-material-ui';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+import SimpleTabs from 'components/common/TabPanel';
+import UOLoader from 'components/common/UOLoader';
+import { PermissionsWithAccessToken } from 'generated/sdk';
+import { useQueriesAndMutationsData } from 'hooks/admin/useQueriesAndMutationsData';
+import useDataApiWithFeedback from 'utils/useDataApiWithFeedback';
+
+const useStyles = makeStyles(theme => ({
+  container: {
+    minHeight: '350px',
+    marginTop: '10px',
+    maxHeight: '550px',
+    overflowY: 'auto',
+    overflowX: 'hidden',
+  },
+  submit: {
+    margin: theme.spacing(3, 0, 2),
+  },
+  error: {
+    color: theme.palette.error.main,
+    marginRight: '10px',
+  },
+  submitContainer: {
+    margin: theme.spacing(2, 0, 2),
+  },
+  darkerDisabledTextField: {
+    '& input': {
+      paddingRight: '2px',
+      color: 'rgba(0, 0, 0, 0.7)',
+      whiteSpace: 'nowrap',
+      overflow: 'hidden',
+      textOverflow: 'ellipsis',
+    },
+  },
+  tabsContainer: {
+    '& > div': {
+      margin: 0,
+      padding: '10px 0',
+      boxShadow: 'none',
+
+      '& [role="tabpanel"] > div': {
+        padding: '0 5px',
+      },
+    },
+  },
+  checkBoxLabelText: {
+    '& label': {
+      width: '100%',
+
+      '& .MuiFormControlLabel-label': {
+        whiteSpace: 'nowrap',
+        overflow: 'hidden',
+        textOverflow: 'ellipsis',
+      },
+    },
+  },
+}));
+
+type FormPermissionsWithAccessToken = {
+  name: string;
+  accessPermissions: string[];
+  accessToken: string;
+};
+
+type CreateUpdateApiAccessTokenProps = {
+  close: (apiAccessTokenAdded: PermissionsWithAccessToken | null) => void;
+  apiAccessToken: PermissionsWithAccessToken | null;
+};
+
+const CreateUpdateApiAccessToken: React.FC<CreateUpdateApiAccessTokenProps> = ({
+  close,
+  apiAccessToken,
+}) => {
+  const classes = useStyles();
+  const { api, isExecutingCall } = useDataApiWithFeedback();
+  const {
+    queriesAndMutations,
+    loadingQueriesAndMutations,
+  } = useQueriesAndMutationsData();
+
+  const normalizeAccessPermissions = (data: string | undefined) => {
+    const permissionsArray: string[] = [];
+
+    if (data) {
+      const parsedPermissions = JSON.parse(data);
+
+      for (const key in parsedPermissions) {
+        if (Object.prototype.hasOwnProperty.call(parsedPermissions, key)) {
+          permissionsArray.push(key);
+        }
+      }
+    }
+
+    return permissionsArray;
+  };
+
+  const initialValues: FormPermissionsWithAccessToken = apiAccessToken
+    ? {
+        ...apiAccessToken,
+        accessToken: `Bearer ${apiAccessToken.accessToken}`,
```

Comment:
Do we need the bearer part here?

Reply:
I think here yes because I would like to show it inside the input to be able just to copy directly as a whole useful token. I have that copy button for the input so you can just copy the token all together and share it with someone who needs to use it.